context.go

```diff
@@ -96,3 +123,77 @@ func (c *Context) Service(name string) Service {
 func (c *Context) String() string {
 	return c.conode.ServerIdentity.String()
 }
+
+var contextData = map[string][]byte{}
+
+// Save takes an identifier and an interface. The interface will be network.Marshaled
+// and written to the filename based on the identifier. An eventual error will be returned.
+func (c *Context) Save(id string, data interface{}) error {
+	buf, err := network.MarshalRegisteredType(data)
+	if err != nil {
+		return err
+	}
+	fname := c.Path(id)
+	if contextDataPath == "" {
+		contextData[fname] = buf
+		return nil
+	}
+	return ioutil.WriteFile(fname, buf, 0660)
+}
+
+// Load takes a filename and returns the network.Unmarshaled data. If an error
+// occurs, the data is nil.
+// The file is read from a service-conode owned directory, so two services have
+// two distinct directories.
+func (c *Context) Load(id string) (interface{}, error) {
+	var buf []byte
+	if contextDataPath == "" {
+		var ok bool
+		buf, ok = contextData[c.Path(id)]
+		if !ok {
+			return nil, errors.New("This exntry doesn't exist")
+		}
+	} else {
+		var err error
+		buf, err = ioutil.ReadFile(c.Path(id))
+		if err != nil {
+			return nil, err
+		}
+	}
+	_, ret, err := network.UnmarshalRegistered(buf)
+	return ret, err
+}
+
+// Path returns the absolute path to load and save the configuration.
+// The file is chosen as "#{ServerIdentity.Public}_#{ServiceName}_#{id}.bin",
+// so no service and no conode share the same file.
+func (c *Context) Path(id string) string {
+	pub, _ := c.ServerIdentity().Public.MarshalBinary()
+	return path.Join(contextDataPath, fmt.Sprintf("%x_%s_%s.bin", pub,
+		ServiceFactory.Name(c.ServiceID()), id))
+}
+
+// Returns the path to the file for storage/retrieval of the service-state.
+// It tries to come up with a suitable name for MacOS/Linux/Windows and also
+// accepts an override with the environmental-variable "CONODE_SERVICE".
```

Comment:
`CONODE_SERVICE_PATH` ?
Or something that indicates the purpose. 
And also, it should be put into a global variable, since you refer to it multiple times here and in the test.